qa/rpc-tests/p2p-segwit.py

```diff
@@ -302,12 +302,12 @@ def test_unnecessary_witness_before_segwit_activation(self):
         sync_blocks(self.nodes)
 
         # We'll add an unnecessary witness to this transaction that would cause
-        # it to be too large according to IsStandard.
+        # it to be non-standard.
         tx3 = CTransaction()
         tx3.vin.append(CTxIn(COutPoint(tx2.sha256, 0), CScript([p2sh_program])))
         tx3.vout.append(CTxOut(tx2.vout[0].nValue-1000, scriptPubKey))
         tx3.wit.vtxinwit.append(CTxInWitness())
-        tx3.wit.vtxinwit[0].scriptWitness.stack = [b'a'*400000]
+        tx3.wit.vtxinwit[0].scriptWitness.stack = [b'a']
```

Comment:
What do you think of this:
``` 
# We'll add an unnecessary witness to this transaction that would cause 
# it to be non-standard, to test that violating policy with a witness before 
# segwit activation doesn't blind a node to a transaction.  Transactions
# rejected for having a witness before segwit activation shouldn't be added
# to the rejection cache.
```

And then at line 312, add a comment:
```
# Note that this should be rejected for the premature witness reason,
# rather than a policy check, since segwit hasn't activated yet.
```